components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -75,396 +55,238 @@
-            for (String originID : originIds) {
-                Resource resource = getResourceById(originID);
-                // Remove the origin association for the tenant/application.
-                getConfigurationManager().deleteAttribute(CORS_ORIGIN_RESOURCE_TYPE_NAME,
-                        resource.getResourceName(), appId);
-                resource = getResourceById(originID);
-                if (resource.getAttributes().isEmpty()) {
-                    getConfigurationManager().deleteResource(CORS_ORIGIN_RESOURCE_TYPE_NAME,
-                            resource.getResourceName());
-                }
-            }
-        } catch (ConfigurationManagementException e) {
-            throw handleServerException(ERROR_CODE_CORS_DELETE, e, tenantDomain);
-        } finally {
-            FrameworkUtils.endTenantFlow();
-        }
+        // DAO layer throws an exception if CORSApplications cannot be retrieved for the corsOriginId.
+        // i.e The corsOriginId is invalid.
+        return Collections.unmodifiableList(getCORSOriginDAO().getCORSOriginApplications(corsOriginId));
     }
 
     /**
-     * Retrieve the ConfigurationManager instance from the CORSServiceHolder.
-     *
-     * @return ConfigurationManager The ConfigurationManager instance.
+     * {@inheritDoc}
      */
-    private ConfigurationManager getConfigurationManager() {
+    @Override
+    public CORSConfiguration getCORSConfiguration(String tenantDomain) throws CORSManagementServiceException {
 
-        return CORSManagementServiceHolder.getInstance().getConfigurationManager();
+        getTenantId(tenantDomain);
+
+        return getCORSConfigurationDAO().getCORSConfigurationByTenantDomain(tenantDomain);
     }
 
     /**
-     * Configuration Management API returns a ConfigurationManagementException with the error code CONFIGM_00017 when
-     * resource is not found. This method wraps the original method and returns null if the resource is not found.
-     *
-     * @param resourceTypeName Resource type name.
-     * @param resourceName     Resource name.
-     * @return Retrieved resource from the configuration store. Returns {@code null} if the resource is not found.
-     * @throws ConfigurationManagementException
+     * {@inheritDoc}
      */
-    private Resource getResource(String resourceTypeName, String resourceName) throws ConfigurationManagementException {
+    @Override
+    public void setCORSConfiguration(CORSConfiguration corsConfiguration, String tenantDomain)
+            throws CORSManagementServiceException {
 
-        try {
-            return getConfigurationManager().getResource(resourceTypeName, resourceName);
-        } catch (ConfigurationManagementException e) {
-            if (e.getErrorCode().equals(ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode())) {
-                return null;
-            } else {
-                throw e;
-            }
-        }
+        getTenantId(tenantDomain);
```

Comment:
What's the usage of this?

Reply:
Calling this function with the tenantDomain validates it.

Reply:
I added a separate function for this.